### DISCUSSION TITLE
I don't know error

### DISCUSSION THREAD

**Discussion by breathinh** [community/community#110975] [Programming Help]
### Body

first my html code below:
<!DOCTYPE html>
<html lang="ko">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>점검 양식</title>
    <style>
        body {
            font-family: Arial, sans-serif;
            margin: 0;
            padding: 0;
            background: #f7f7f7;
            display: flex;
            justify-content: center;
            align-items: center;
            min-height: 100vh;
        }
        .container {
            width: 90%;
            max-width: 600px;
            margin: auto;
        }
        input, select, textarea {
            width: 100%;
            padding: 8px;
            margin: 8px 0;
            display: inline-block;
            border: 1px solid #ccc;
            border-radius: 4px;
            box-sizing: border-box;
        }
        .submit-button {
            width: 100%;
            background-color: #4CAF50;
            color: white;
            padding: 14px 20px;
            margin: 8px 0;
            border: none;
            border-radius: 4px;
            cursor: pointer;
        }
        .submit-button:hover {
            background-color: #45a049;
        }
        /* 모바일 장치에 대한 스타일 */
        @media screen and (max-width: 768px) {
            .container {
                width: 100%;
                margin: 0;
                padding: 0;
                min-height: 100vh; /* 최소 높이를 뷰포트 높이의 100%로 설정 */
            }
            input, select, textarea, .submit-button {
                padding: 12px;
                margin: 10px 0;
            }
            body {
                overflow-x: hidden; /* 가로 스크롤 방지 */
            }
        }
    </style>
</head>
<body>
    <div class="container">
        <h2>점검 양식</h2>
        <form id="inspection-form" action="https://script.google.com/macros/s/AKfycbwsiZk90-yW5X5944F9ICWKX0ktHUNVnpKtX_TvFMkgbNEKnIprb72n4Nx6HsbEi4RM2A/exec" method="post" enctype="multipart/form-data">

            <label for="building-field">동</label>
            <input type="text" id="building" name="building" placeholder="동을 입력하세요">

            <label for="unit-field">호수</label>
            <input type="text" id="unit" name="unit" placeholder="호수를 입력하세요">

            <label for="type-field">타입</label>
            <input type="text" id="type" name="type" placeholder="타입을 입력하세요">

            <div class="form-group">
                <label for="date">일자</label>
                <input type="date" id="date" name="date">
            </div>
            <div class="form-group">
                <label for="code">공종</label>
                <select id="code" name="code">
                    <option value="">선택하세요</option>
                    
                </select>
            </div>
            <div class="form-group">
                <label for="location">내용</label>
                <textarea id="location" name="location" rows="4"></textarea>
            </div>
            <div class="form-group">
                <label for="details">지적사진</label>
                <input type="file" id="details" name="details" placeholder="파일 선택 또는 파일 이름">
            </div>
            <div class="form-group">
                <button type="submit" class="form-button">제출</button>
                <button type="reset" class="form-button">취소</button>
            </div>
        </form>
    </div>

    <script>
        document.addEventListener('DOMContentLoaded', function() {
            // 폼의 필드 초기화를 위해 DOM 요소를 가져옵니다.
            const building = document.getElementById('building');
            const unit = document.getElementById('unit');
            const type = document.getElementById('type');
            const form = document.getElementById('inspection-form');

            // 폼 제출 이벤트 리스너를 설정합니다.
            form.addEventListener('submit', function(e) {
                e.preventDefault(); // 기본 폼 제출을 방지합니다.

                // 동, 호수, 타입 필드를 제외하고 폼을 리셋하는 함수입니다.
                function resetFormExceptBuildingUnitType() {
                    // 폼 내의 모든 input, select, textarea 요소를 가져옵니다.
                    const inputs = form.querySelectorAll('input, select, textarea');
                    inputs.forEach(input => {
                        // 동, 호수, 타입 필드는 리셋하지 않습니다.
                        if (input !== building && input !== unit && input !== type) {
                            input.value = '';
                        }
                    });
                }
                
                // 페이지 리디렉션을 수행할 URL
                var redirectTo = "https://tetserver-c644b.web.app/?building=" + encodeURIComponent(building.value) + "&unit=" + encodeURIComponent(unit.value) + "&type=" + encodeURIComponent(type.value);

                // 페이지 리디렉션 함수
                function redirect() {
                    window.location.replace(redirectTo);
                }

                // 폼을 리셋한 후에 리디렉션 함수 호출
                resetFormExceptBuildingUnitType();
                redirect();
            });
        });
    </script>
</body>
</html>

and my apps script code below:
function doPost(e) {
  var ss = SpreadsheetApp.openById('1f4EQWGGhjNzup3lys0nkUo6BmCMsNAxIUcxD2R36rg4');
  var sheet = ss.getSheetByName('시트1');

  var formData = {
    building: e.parameter.building,
    unit: e.parameter.unit,
    type: e.parameter.type,
    date: e.parameter.date,
    code: e.parameter.code,
    location: e.parameter.location,
    details: e.parameter.details
  };
  
  // 스프레드시트에 데이터를 추가하는 로직
  var newRow = sheet.appendRow([formData.building, formData.unit, formData.type, formData.date, formData.code, formData.location, formData.details]);
  
  // 리디렉션할 URL을 구성합니다. 동, 호수, 타입 정보를 쿼리 파라미터로 포함합니다.
  var redirectUrl = 'https://tetserver-c644b.web.app/?building=' + encodeURIComponent(formData.building) + '&unit=' + encodeURIComponent(formData.unit) + '&type=' + encodeURIComponent(formData.type);
  
  // 사용자를 리디렉션 URL로 보내는 HTML 응답을 생성합니다.
  var htmlOutput = HtmlService.createHtmlOutput('Redirecting...<script>window.location="' + redirectUrl + '";</script>');
  return htmlOutput.setXFrameOptionsMode(HtmlService.XFrameOptionsMode.ALLOWALL);
}

I want that redirectUrl is correct response but after submission then change https://script.google.com/macros/s/AKfycbyzLPEvalOb3SGBJj4Ae1FGCuMWYSIWgQZ52cdyTUwpD1k06068lmKvdIlPwARD8zCrUw/exec 
please help me

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by LucianoCassarini** [comment]
Hello!

You could try modifying your HTML script to handle the JSON response and perform the redirect after submitting the form.

Make sure you have correctly set the redirectUrl variable in your Google Apps script and are handling the response in your HTML script. With these modifications, you should be able to dynamically change the redirect URL after submitting the form.

```
<script>
  document.addEventListener('DOMContentLoaded', function () {
    const form = document.getElementById('inspection-form');

    form.addEventListener('submit', function (e) {
      e.preventDefault();

      // Obtener los datos del formulario y enviarlos al script de Google Apps
      google.script.run.withSuccessHandler(handleSuccess).doPost(new FormData(form));
    });

    // Función para manejar la respuesta exitosa del script de Google Apps
    function handleSuccess(response) {
      if (response.success) {
        // Redirigir a la URL proporcionada en la respuesta JSON
        window.location.href = response.redirectUrl;
      } else {
        // Manejar cualquier error que pueda ocurrir
        console.error('Error en el script de Google Apps');
      }
    }
  });
</script>
```
